packages/app/src/providers/hsds/utils.ts

```diff
@@ -90,19 +102,25 @@ function convertHsdsCompoundType(
     assertHsdsNumericType(realType);
     assertHsdsNumericType(imagType);
 
-    return {
-      class: DTypeClass.Complex,
-      realType: convertHsdsNumericType(realType),
-      imagType: convertHsdsNumericType(imagType),
-    };
+    return cplxType(
+      convertHsdsNumericType(realType),
+      convertHsdsNumericType(imagType),
+    );
   }
 
-  return {
-    class: DTypeClass.Compound,
-    fields: Object.fromEntries(
+  return compoundType(
+    Object.fromEntries(
       hsdsType.fields.map((v) => [v.name, convertHsdsType(v.type)]),
     ),
-  };
+  );
+}
+
+function convertHsdsEnumType(hsdsType: HsdsEnumType): EnumType | BooleanType {
+  const { base, mapping } = hsdsType;
+  assertHsdsNumericType(base);
+
+  const type = enumType(convertHsdsNumericType(base), mapping);
+  return isBoolEnumType(type) ? boolType() : type; // booleans stored as enums by h5py
```

Comment:
I've also added a utility to abstract away from the providers the check of whether an enum dtype corresponds to the boolean dtype.